codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -554,33 +583,37 @@ final void weight(short weight) {
          */
         final void takeChild(DefaultStream child, boolean exclusive, List<ParentChangedEvent> events) {
             DefaultStream oldParent = child.parent();
-            events.add(new ParentChangedEvent(child, oldParent));
-            notifyParentChanging(child, this);
-            child.parent = this;
 
-            if (exclusive && !children.isEmpty()) {
-                // If it was requested that this child be the exclusive dependency of this node,
-                // move any previous children to the child node, becoming grand children of this node.
-                for (DefaultStream grandchild : removeAllChildren().values()) {
-                    child.takeChild(grandchild, false, events);
+            if (oldParent != this) {
+                events.add(new ParentChangedEvent(child, oldParent));
+                notifyParentChanging(child, this);
+                child.parent = this;
+                // We need the removal operation to happen first so the prioritizableForTree for the old parent to root
+                // path is updated with the correct child.prioritizableForTree() value.
+                if (oldParent != null && oldParent.children.remove(child.id()) != null) {
```

Comment:
>> if oldParent != null, isn't it an error condition if removal of the child from oldParent is null?

>Seems like you are saying "if `oldParent == null` isn't this an error".  This is what my `I don't think so` applied to.

Ah ok, I think I understand your confusion now :).  So looking at this line:

```java
if (oldParent != null && oldParent.children.remove(child.id()) != null)
```

My original question was: how can `oldParent.children.remove(child.id())` ever be `null`? In fact, it seems like it should be an error condition as this would imply an inconsistency in the tree ... or am I missing something?